java/ql/src/experimental/Security/CWE/CWE-094/InsecureBeanValidation.ql

```diff
@@ -0,0 +1,76 @@
+/**
+ * @name Insecure Bean validation
+ * @description User-controlled data may be evaluated as a Java EL expressions, leading to arbitrary code execution.
+ * @kind path-problem
+ * @problem.severity error
+ * @precision high
+ * @id java/unsafe-eval
+ * @tags security
+ *       external/cwe/cwe-094
+ */
+
+import java
+import semmle.code.java.dataflow.TaintTracking
+import semmle.code.java.dataflow.FlowSources
+import DataFlow::PathGraph
+
+class BeanValidationSource extends RemoteFlowSource {
```

Comment:
Is this a remote flow source that ought to apply to all security queries or only this one? Conversely, does this query intend to use all remote flow sources as sources or just the `BeanValidationSource`?

Reply:
In theory, the validated bean properties could reach a different sink (eg: a SQLi one) so this source need to be accounted for all security queries. Ideally the source should not be necessary but that would require taint steps like the ones in @adityasharad version of the query. That version of the query missed the new issues so I would stick to this source and treat it as a generic remote source. 

@adityasharad are you ok with this? otherwise we can try to debug your query and see why its not working for the new cases.